classes/controllers/grid/issues/IssueGridHandler.inc.php

```diff
@@ -602,26 +606,25 @@ public function publishIssue($args, $request)
 
         // Send a notification to associated users if selected and context is publishing content online with OJS
         if ($request->getUserVar('sendIssueNotification') && $context->getData('publishingMode') != \APP\journal\Journal::PUBLISHING_MODE_NONE) {
-            $notificationManager = new NotificationManager();
-            $notificationUsers = [];
-            $userGroupDao = DAORegistry::getDAO('UserGroupDAO'); /** @var UserGroupDAO $userGroupDao */
-            $allUsers = $userGroupDao->getUsersByContextId($contextId);
-            while ($user = $allUsers->next()) {
-                if ($user->getDisabled()) {
-                    continue;
-                }
-                $notificationUsers[] = ['id' => $user->getId()];
-            }
-            foreach ($notificationUsers as $userRole) {
-                $notificationManager->createNotification(
-                    $request,
-                    $userRole['id'],
-                    Notification::NOTIFICATION_TYPE_PUBLISHED_ISSUE,
+            /** @var NotificationSubscriptionSettingsDAO $notificationSubscriptionSettingsDao */
+            $notificationSubscriptionSettingsDao = DAORegistry::getDAO('NotificationSubscriptionSettingsDAO');
+
+            $userIdsToMail = $notificationSubscriptionSettingsDao->getSubscribedUserIds(
+                [NotificationSubscriptionSettingsDAO::BLOCKED_EMAIL_NOTIFICATION_KEY],
+                [Notification::NOTIFICATION_TYPE_PUBLISHED_ISSUE],
+                [$contextId]
+            );
+
+            foreach ($userIdsToMail->chunk(Mailer::BULK_EMAIL_SIZE_LIMIT) as $mailUserIds) {
```

Comment:
Not something to change, but just heads up you can use the collection methods to chain functions in neat ways:

```php
$jobs = $userIdsToMail
    ->chunk(Mailer::BULK_EMAIL_SIZE_LIMIT)
    ->map(fn($userIds) => new IssuePublishedMailUsers(
        $userIds,
        $contextId,
        $request->getUser()->getId(),
        Locale::getLocale()
    ));
```